chromedriver artifact arch has changed to `mac_arm64` from `mac64_m1`

Looks like the latest chromedriver from google now has `mac_arm64` instead of `mac64_m1` as the arch. This can be seen searching for `106.0.5249.61/chromedriver_` here https://chromedriver.storage.googleapis.com/

```
❯ npm install chromedriver --no-save --detect_chromedriver_version

npm ERR! code 1
npm ERR! path /Users/svarozian/Projects/veza/cookieai-core/node_modules/chromedriver
npm ERR! command failed
npm ERR! command sh -c -- node install.js
npm ERR! Your Chrome version is 106.0.5249.61
npm ERR! Finding Chromedriver version.
npm ERR! Chromedriver version is 106.0.5249.61.
npm ERR! Compatible ChromeDriver version is 106.0.5249.61
npm ERR! Current existing ChromeDriver binary is unavailable, proceeding with download and extraction.
npm ERR! Downloading from file:  https://chromedriver.storage.googleapis.com/106.0.5249.61/chromedriver_mac64_m1.zip
npm ERR! Saving to file: /var/folders/f6/2pvgz0ls67d_dstpwmmfpj4h0000gp/T/106.0.5249.61/chromedriver/chromedriver_mac64_m1.zip
npm ERR! Error status code: 404
npm ERR! <?xml version='1.0' encoding='UTF-8'?><Error><Code>NoSuchKey</Code><Message>The specified key does not exist.</Message><Details>No such object: chromedriver/106.0.5249.61/chromedriver_mac64_m1.zip</Details></Error>
npm ERR! ChromeDriver installation failed Error: Error with http(s) request: AxiosError: Request failed with status code 404
npm ERR!     at requestBinary (/Users/svarozian/Projects/veza/cookieai-core/node_modules/chromedriver/install.js:274:11)
npm ERR!     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
npm ERR!     at async downloadFile (/Users/svarozian/Projects/veza/cookieai-core/node_modules/chromedriver/install.js:117:5)
npm ERR!     at async install (/Users/svarozian/Projects/veza/cookieai-core/node_modules/chromedriver/install.js:66:7)

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/svarozian/.npm/_logs/2022-09-28T17_53_36_774Z-debug-0.log
```

--- a/install.js
+++ b/install.js
@@ -16,6 +16,7 @@ const extractZip = require('extract-zip');
 const { getChromeVersion } = require('@testim/chrome-version');
 const HttpsProxyAgent = require('https-proxy-agent');
 const getProxyForUrl = require("proxy-from-env").getProxyForUrl;
+const { compareVersions } = require('compare-versions');
 
 const finishedAsync = promisify(finished);
 
@@ -31,12 +32,12 @@ const configuredfilePath = process.env.npm_config_chromedriver_filepath || proce
 
 // adapt http://chromedriver.storage.googleapis.com/
 cdnUrl = cdnUrl.replace(/\/+$/, '');
-const platform = validatePlatform();
 const detect_chromedriver_version = process.env.npm_config_detect_chromedriver_version || process.env.DETECT_CHROMEDRIVER_VERSION;
 const include_chromium = (process.env.npm_config_include_chromium || process.env.INCLUDE_CHROMIUM) === 'true';
 let chromedriver_version = process.env.npm_config_chromedriver_version || process.env.CHROMEDRIVER_VERSION || helper.version;
 let chromedriverBinaryFilePath;
 let downloadedFile = '';
+let platform = '';
 
 (async function install() {
   try {
@@ -57,6 +58,7 @@ let downloadedFile = '';
         await getChromeDriverVersion(getRequestOptions(`${cdnUrl}/LATEST_RELEASE_${majorVersion}`));
       }
     }
+    platform = validatePlatform();
     const tmpPath = findSuitableTempDirectory();
     const chromedriverBinaryFileName = process.platform === 'win32' ? 'chromedriver.exe' : 'chromedriver';
     chromedriverBinaryFilePath = path.resolve(tmpPath, chromedriverBinaryFileName);
@@ -341,7 +343,11 @@ function fixFilePermissions() {
 
 function getMacOsRealArch() {
   if (process.arch === 'arm64' || isEmulatedRosettaEnvironment()) {
-    return 'mac64_m1';
+    if (compareVersions(chromedriver_version, '106.0.5249.61') < 0) {
+      return 'mac64_m1';
+    }
+
+    return 'mac_arm64'
   }
 
   if (process.arch === 'x64') {

--- a/package.json
+++ b/package.json
@@ -28,6 +28,7 @@
   "dependencies": {
     "@testim/chrome-version": "^1.1.3",
     "axios": "^0.27.2",
+    "compare-versions": "^5.0.1",
     "del": "^6.1.1",
     "extract-zip": "^2.0.1",
     "https-proxy-agent": "^5.0.1",

